homeassistant/components/zwave_js/sensor.py

```diff
@@ -296,6 +300,40 @@ def native_unit_of_measurement(self) -> str | None:
 class ZWaveNumericSensor(ZwaveSensorBase):
     """Representation of a Z-Wave Numeric sensor."""
 
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        client: ZwaveClient,
+        info: ZwaveDiscoveryInfo,
+        entity_description: SensorEntityDescription,
+        unit_of_measurement: str | None = None,
+    ) -> None:
+        """Initialize a ZWaveNumericSensor entity."""
+        super().__init__(
+            config_entry, client, info, entity_description, unit_of_measurement
+        )
+        self._scale = self.info.primary_value.metadata.cc_specific.get(
+            CC_SPECIFIC_SCALE
+        )
+
+    @callback
+    def on_value_update(self) -> None:
+        """Handle scale changes for this value on value updated event."""
+        # If the scale hasn't changed, we can return early
+        if self._scale == (
```

Comment:
I don't think this guard is needed anymore. It made more sense when we were trying to catch the events within the class instance but I think now we can just set the unit every time. Thoughts?

Reply:
Sounds good.